### PR TITLE
[ROCm] enforce ROCM_VERSION >= 6.0

### PR DESCRIPTION
**Pull request by jeffdaily**
Remove any code relying on ROCM_VERSION < 6.0.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @EikanWang @jgong5

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks for the cleanup!
What is the process that we're using to share the schedule of supported versions for rocm with users? 
Should we add a column to this https://github.com/pytorch/pytorch/blob/main/RELEASE.md#release-compatibility-matrix ?

**Comment by jeffdaily** [comment]
> Thanks for the cleanup! What is the process that we're using to share the schedule of supported versions for rocm with users? Should we add a column to this https://github.com/pytorch/pytorch/blob/main/RELEASE.md#release-compatibility-matrix ?

Great suggestion.  I will add a column for ROCm.  The practice up until this PR was to be backward compatible up to 2 prior ROCm releases.  CI only tests the current ROCm.  Nightly builds cover current ROCm and N-1.  Internally we validate latest ROCm + latest pytorch when we make a release.  So in practice no one really verifies that pytorch will build on ROCm versions more than 1 release back.

Would you be able to add a reviewer that can do the Meta-internal phabricator import?  I'm concerned these changes might break internal ROCm builds.  That has happened with another of my PRs.  I would like to avoid revert churn.

**Comment by albanD** [comment]
Sounds great!
Happy to review a PR putting these expectations and plan in RELEASE.md !

Will check how that goes internally!

**Comment by albanD** [comment]
Ah! That explains why it was failing internally!

**Comment by albanD** [comment]
Ok, it imported cleanly, waiting for CI status and will report back here.

**Comment by jeffdaily** [comment]
> Sounds great! Happy to review a PR putting these expectations and plan in RELEASE.md !
> 
> Will check how that goes internally!

Done.  Added ROCm compatibility column to the RELEASE.md table.

File: aten/src/ATen/native/cuda/Blas.cpp [hunk 1]
```diff
@@ -770,7 +766,7 @@ Tensor& _int_mm_out_cuda(const Tensor& self, const Tensor& mat2, Tensor& result)
 
   TORCH_CHECK(result.is_contiguous(), "Expected result to be contiguous.");
 
-#if (!defined(USE_ROCM) && defined(CUDA_VERSION) && (CUDA_VERSION >= 11070)) || (defined(USE_ROCM) && (ROCM_VERSION >= 60000))
+#if (!defined(USE_ROCM) && defined(CUDA_VERSION) && (CUDA_VERSION >= 11070)) || defined(USE_ROCM)
```
Comment (eqy) [inline aten/src/ATen/native/cuda/Blas.cpp:769]: nit: maybe we can remove the `!defined(USE_ROCM)`

**Review by eqy — APPROVED** [review]

File: aten/src/ATen/native/cuda/Blas.cpp (hunk 1 above)
Comment (jeffdaily) [reply aten/src/ATen/native/cuda/Blas.cpp:769]: Good catch.  Let me make a pass through all code again looking for this anti-pattern.

Comment (jeffdaily) [reply aten/src/ATen/native/cuda/Blas.cpp]: Fixed.  Made a pass through the code and it all adheres to https://pytorch.org/docs/stable/notes/hip.html#cuda-api-to-hip-api-mappings-in-c now.

**Comment by albanD** [comment]
@pytorchbot merge

All good internally, unlinked the diff and landing here!

**Pull request closed without merging**